dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobConfiguration.java

```diff
@@ -341,7 +341,10 @@ public boolean isUsedInQueue() {
   }
 
   public boolean isRunOnce() {
-    return cronExpression == null && delay == null && queueName == null;
+    return schedulingType == SchedulingType.ONCE_ASAP
+        && cronExpression == null
+        && delay == null
+        && queueName == null;
```

Comment:
FYI: the reason to check all the fields is so we do not exclude scheduled jobs which got executed manually so their `schedulingType` shortly switches to `ONCE_ASAP` as well but they still have a `cronExpression.